configureoptimization.cmake

```diff
@@ -0,0 +1,22 @@
+if(WIN32)
+    add_compile_options($<$<CONFIG:Debug>:/Od>)
+    add_compile_options($<$<CONFIG:Checked>:/O1>)
+    add_compile_options($<$<CONFIG:Release>:/Ox>)
+    add_compile_options($<$<CONFIG:RelWithDebInfo>:/O2>)
+elseif(CLR_CMAKE_PLATFORM_UNIX)
+    set(CLR_CMAKE_ARM_OPTIMIZATION_FALLBACK OFF)
+    if(CLR_CMAKE_TARGET_ARCH STREQUAL "arm" OR CLR_CMAKE_TARGET_ARCH STREQUAL "armel")
+        if(("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang") AND (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.9))
+            set(CLR_CMAKE_ARM_OPTIMIZATION_FALLBACK ON)
+        endif()
+    endif()
+    add_compile_options($<$<CONFIG:Debug>:-O0>)
+    if (CLR_CMAKE_ARM_OPTIMIZATION_FALLBACK)
+        add_compile_options($<$<NOT:$<CONFIG:Debug>>:-O1>)
```

Comment:
This is not correct. It should be changing the Checked, Release and RelWithDebInfo configurations, not the Debug one.
I am actually surprised the conditional expressions work for Unix generators, we use them only for the multi-generator (for Windows) at other places.

Reply:
This is changing all of the non-Debug configurations. The generator expressions work for all CMake generators, but for non-multi-generators, you specify the configuration with `CMAKE_BUILD_TYPE` (which is what we do).

Reply:
Oops,  I am sorry, my eyes have missed the `NOT` in the condition.